### PR TITLE
Go gutters

### PR DESCRIPTION
**Pull request by ctwarog-ross**
## 📖 Description
Coverage gutters now works in Go SDK.

---

## ✅ Definition of Done (DoD)
- [x] Feature is implemented and works as expected
- [x] Edge cases are handled
- [x] Tests are added or updated (if applicable)
- [x] Documentation is updated (if needed)
- [ ] Code is reviewed and approved

---

## 🛠️ Implementation Notes
gcov2lcov converts coverage.out to lcov

---

## 🔗 Related Issues / Links

---
## 📝 Additional Context

---

## 🚧 Risks / Open Questions

---
## ✨ Updates
<!-- (Optional) List summarizing changes made when editing the pull request>

### CODE REVIEW

File: .vscode/settings.json
Comment (abrown-RV) [inline .vscode/settings.json:1]: breaks cpp coverage, we were running with the default value for
```
	"coverage-gutters.coverageFileNames": [
		"lcov.info",
		"cov.xml",
		"coverage.xml",
		"cobertura.xml",
		"jacoco.xml",
		"coverage.cobertura.xml"
	]
```
But this overwrites the default, so it looses the cpp coverage files that were working with the default. cpp coverage goes to `coverage.xml` and `coverage.info` so I guess we just need the `coverage.xml` from the default, but idk, maybe just put the whole thing in?

Either that or we start organizing the coverage into subdirs? idk no convinced that's needed.

File: .vscode/settings.json [hunk 1]
```diff
@@ -149,5 +149,13 @@
...
         "typeindex": "cpp"
     },
-    "java.compile.nullAnalysis.mode": "disabled"
+    "java.compile.nullAnalysis.mode": "disabled",
+    "coverage-gutters.coverageFileNames": [
+        "coverage/go_coverage.lcov",
+        "coverage.lcov",
+        "lcov.info"
+    ],
+    "coverage-gutters.coverageBaseDir": "**",
```
Comment (abrown-RV) [inline .vscode/settings.json]: `"coverage-gutters.coverageBaseDir": "**"` is the default value, I just don't like hardcoding the default if we want the same

File: .gitignore [hunk 2]
```diff
@@ -96,4 +96,4 @@ scripts/reports/
 #Go
 *__debug_bin*
 go.work*
-*coverage.out
+*coverage.out
```
Comment (abrown-RV) [inline .gitignore]:
```suggestion
*coverage.out

```

**Review by abrown-RV — CHANGES_REQUESTED** [review]
settings.json needs a touch, but thanks so much for doing this, makes review so much nicer! :D

**Review by abrown-RV — APPROVED** [review]
